__main__.py

```diff
@@ -0,0 +1,93 @@
+# Welcome to the WEAVER. This module......
+#
+#
+# Copyright (C) 2015 Weecology
+#
+# This module is part of weaver and is released under
+# the MIT License: http://www.opensource.org/licenses/mit-license.php
+
+from weaver.engines import engine_list
+from weaver.lib.engine import Engine
+from weaver import VERSION
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy import *
+import os
+import platform
+import sys
+import mysql.connector
+import psycopg2
+from weaver.lib.get_opts import parser
+import json
+from sqlalchemy import create_engine, Column, String, Integer, MetaData, Table
+from sqlalchemy.orm import mapper, create_session
+import csv
+import time
+
+
+import os
+import platform
+import sys
+
+# sys removes the setdefaultencoding method at startup; reload to get it back
+
+reload(sys)
+if hasattr(sys, 'setdefaultencoding'):
+    # set default encoding to latin-1 to avoid ascii encoding issues
+    sys.setdefaultencoding('latin-1')
+
+
+def main():
+    """
+    main function the weaver module
+    """
+
+    # if no command line args are passed, launch GUI
+    if len(sys.argv) == 1 or (len(sys.argv) > 1 and sys.argv[1] == 'gui'):
```

Comment:
I think I'm preferential to a separate command to launch the GUI like `retriever-gui`. See https://github.com/weecology/retriever/issues/323 for more discussion of what I'm thinking.